src/components/Course.jsx

```diff
@@ -0,0 +1,59 @@
+// ============================================================================================
+// IMPORTS
+// ============================================================================================
+
+import PropTypes from "prop-types";
+import Module from "./Module.jsx";
+import Assessment from "./Assessment.jsx";
+
+import "./Course.css";
+
+// ============================================================================================
+// COMPONENT DEFINITION
+// ============================================================================================
+
+/*********************************************************************************************/
+
+function Course({ courseData }) {
+  return (
+    <div id="Course">
+      <h1>{courseData.title}</h1>
+      <p>
+        <aside style={{ float: "inline-end" }}>
+          <b>Instructor:</b>
+          <br />
+          {courseData.instructor}
+        </aside>
+
+        {courseData.description}
+      </p>
+
+      <section style={{ clear: "both" }}>
+        {courseData.components.map((component, index) => {
+          return (
+            <>
+              {index === 0 || <hr />}
+              <article key={index}>
+                {component?.questions ? (
+                  <Assessment assessmentData={component} />
+                ) : (
+                  <Module moduleData={component} />
+                )}
+              </article>
+            </>
+          );
+        })}
+      </section>
+    </div>
+  );
+}
+
+Course.propTypes = {
+  courseData: PropTypes.object.isRequired
+};
```

Comment:
Course.propTypes = {
  courseData: PropTypes.shape({
    title: PropTypes.string.isRequired,
    instructor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    components: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        questions: PropTypes.array,
      }).isRequired
    ).isRequired,
  }).isRequired,
};

Reply:
This is a very good specification, and I agree that `Course.propTypes` needs to be more specific.  I'll implement this when the database schema is more stable.